src/interaction/FixedLocalTupleComListInteractionTemplate.hpp

```diff
@@ -92,18 +117,20 @@ namespace espressopp {
 		pid_list = new int[num_of_subchain];
 
 		// Check the particle id in FixedLocalTuple
-		for (FixedLocalTupleList::TupleList::Iterator it(*fixedtupleList); it.isValid(); ++it) {
-		    Particle *p = it->first;
-		    pid_list[pidK] = p->id();
-		    
-		    for(long unsigned int i = 0; i < pidK; i++) {
-			if (pid_list[i]/N_Constrain == pid_list[pidK]/N_Constrain) {
-			    std::stringstream msg;
-			    msg << "ERROR: Particle ID is redundant\n";
-			    err.setException(msg.str());
+		if (!fixedtupleList->empty()) {
```

Comment:
Why do you need everywhere to check if the list empty? isValid() on empty container should return `false` so the for-loop will not be executed at all.